### PR TITLE
Added a Range(min, max) function

### PR DESCRIPTION
**Pull request by alessandrousseglioviretta**
The new function allows ranging over indexed fields.

### CODE REVIEW

**Comment by asdine** [comment]
Thank you @alessandrousseglioviretta 
I like it, `Range` can be a great addition to Storm.

I would rather use this signature for the public `Range` method though:
```go
Range(fieldName string, min interface{}, max interface{}, to interface{}) error
```
The `min` and `max` params are gonna be converted to bytes with the `toBytes()` function [here](https://github.com/asdine/storm/pull/5/files#diff-af8f5b33ad6e67e946d8f0d47588d5aaR53).
Usage example:
```go
db.Range("Amounts", 100, 200, &items)
db.Range("CreationDate", time.Now().Add(-time.Duration(10)*time.Hours), time.Now(), &users)
...
```

**Comment by alessandrousseglioviretta** [comment]
Yep, I agree. I already pushed the changes to my repo. I'm not very familiar with the normal process: should I make a new pull request?

Also, I added a FindByPrimaryKey helper method.

**Comment by asdine** [comment]
I don't agree with `FindByPrimaryKey`, i think it is irrelevant to this PR and it is basically `One("ID", 10, &user)`. If you think it is necessary to have one,  please open an issue so we can discuss about that.

Also can you add more tests please, with other types (time.Time, ints, ...) and zero values for `min` and `max`, so we are sure that it works with any type?

**Comment by alessandrousseglioviretta** [comment]
Agreed, `FindByPrimaryKey` is just a shortcut to avoid 2 reads instead of one which, I suppose, could be noticeable at high read rates: what's your take on that? Do you have a way to avoid merging the 2 corresponding files?

About the tests: I'll try to implement them soon.

**Comment by asdine** [comment]
I can't exclude a file from the PR unfortunately but you can remove the commit using this method http://stackoverflow.com/questions/20765030/git-remove-a-commit-from-pull-request, so we can have a clean commit history.

Your point on the 2 reads is interesting. Can you please open an issue [here](https://github.com/asdine/storm/issues/new) so we can discuss about it?

File: find_by_primary_key.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+			return fmt.Errorf("bucket %s not found", bucketName)
+		}
+
+		val, err := toBytes(value)
+		if err != nil {
+			return err
+		}
+
+		raw := bucket.Get(val)
+		err = json.Unmarshal(raw, &to)
```
Comment (bep) [inline find_by_primary_key.go]: Needs to use the newly introduced `s.Codec.Decode` and similar below.

Comment (alessandrousseglioviretta) [inline find_by_primary_key.go]: Could you please elaborate? Instead of `Un/Marshal`?

Comment (bep) [inline find_by_primary_key.go]: Rebase against master and it should be pretty clear.

Comment (alessandrousseglioviretta) [inline find_by_primary_key.go]: @bep ok, I've seen the wrapper...
@asdine I've hit the limits of my git knowledge, I didn't manage to uncommit...

**Comment by alessandrousseglioviretta** [comment]
I might have just managed to remove the unwished commits... please check

**Comment by asdine** [comment]
There are still some merge conflicts, and the history isn't clean.
I suggest opening a new pull request, from a clean, up to date branch, and to copy the `Range` code in a single commit. I believe it will be faster

**Pull request closed without merging**